src/test/java/com/algorand/algosdk/util/TestEncoder.java

```diff
@@ -131,4 +137,38 @@ public void testDecodeUint64() {
                     .isInstanceOf(IllegalArgumentException.class)
                     .hasMessage("Length of byte array is invalid");
     }
+
+    @ParameterizedTest
+    @CsvSource({
+            "pay, pay",
+            "blah, ",
+            "blah, ''"
+    })
+    public void testTransactionEnum(String type, String expected) throws IOException {
+        String txnString = "{\"type\": \"" + type + "\"}";
+        Transaction txn = Encoder.decodeFromJson(txnString, Transaction.class);
+        if (expected != null) {
+            Transaction.Type enumValue = Transaction.Type.forValue(expected);
+            assertThat(enumValue).isEqualTo(txn.type);
+        } else {
+            assertThat(Transaction.Type.Default).isEqualTo(txn.type);
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "sig, sig",
+            "blah, ",
+            "blah, ''"
+    })
+    public void testHttpModelEnum(String type, String expected) throws IOException {
```

Comment:
@winder Optional - feel welcomed to proceed as is:  I'd feel more comfortable if `testHttpModelEnum` (or a new test) performs round-trip serialization.

The intent is to prove `UNKNOWN` serialization is consistent with the corresponding Jackson annotation configuration.

Reply:
You just reminded me of a utility that does the round trip encoding test for json and msgpack, I updated the test to include that.

It's not perfect though, since this change is specifically making it so that the JSON to object conversion loses information.

Reply:
@winder Thanks for bringing the feedback.  I've taken note of the mentioned utility method - Resolving.